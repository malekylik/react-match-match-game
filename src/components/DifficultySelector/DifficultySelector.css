.start-difficulty-input {
    display: flex;
    justify-content: space-between;
}
