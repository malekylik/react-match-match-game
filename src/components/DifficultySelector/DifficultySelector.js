import React from 'react';
import PropTypes from 'prop-types';

import './DifficultySelector.css';

const DifficultySelector = ({ id, labelText, onClick, defaultChecked = false }) => (
    <p className="start-difficulty-input">
        <label htmlFor={id}>{labelText}</label>
        <input id={id} type="radio" name="difficulty" onClick={onClick} defaultChecked={defaultChecked} />
    </p>
);

DifficultySelector.propTypes = {
    id: PropTypes.string.isRequired,
    labelText: PropTypes.string.isRequired,
    defaultChecked: PropTypes.bool,
    onClick: PropTypes.func
};

export default DifficultySelector;
