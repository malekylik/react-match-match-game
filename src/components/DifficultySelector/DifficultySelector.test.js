import React from 'react';

import { shallow } from 'enzyme';

import DifficultySelector from './DifficultySelector';

describe('DifficultySelector', () => {
    it('Should match snapshot', () => {
        expect(shallow(
            <DifficultySelector
                id="test1"
                labelText="test1"
                onClick={jest.fn()}
                defaultChecked={true} />)).toMatchSnapshot();
    });
});
