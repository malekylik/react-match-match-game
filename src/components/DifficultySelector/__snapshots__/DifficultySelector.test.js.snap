// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DifficultySelector Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DifficultySelector
    defaultChecked={true}
    id="test1"
    labelText="test1"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test1"
        >
          test1
        </label>,
        <input
          defaultChecked={true}
          id="test1"
          name="difficulty"
          onClick={[MockFunction]}
          type="radio"
        />,
      ],
      "className": "start-difficulty-input",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test1",
          "htmlFor": "test1",
        },
        "ref": null,
        "rendered": "test1",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "defaultChecked": true,
          "id": "test1",
          "name": "difficulty",
          "onClick": [MockFunction],
          "type": "radio",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test1"
          >
            test1
          </label>,
          <input
            defaultChecked={true}
            id="test1"
            name="difficulty"
            onClick={[MockFunction]}
            type="radio"
          />,
        ],
        "className": "start-difficulty-input",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test1",
            "htmlFor": "test1",
          },
          "ref": null,
          "rendered": "test1",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultChecked": true,
            "id": "test1",
            "name": "difficulty",
            "onClick": [MockFunction],
            "type": "radio",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
